### PR TITLE
[AMORO-2255] Fix BasicIcebergTable's name and serialization ID

### LINKED ISSUE #2255: [Bug]: Incorrect name of BasicIcebergTable
### What happened?

The `name()` of BasicIcebergTable is not correctly implemented and returns the default `toString()` which prints the memory address.

Note that the ArticTable uses Java Object serialization and hasn't set a serialization ID, which would block changes to the BasicIcebergTable class, thus I added a commit to pin the serialization ID before overriding the `name()` method.

### Affects Versions

0.6.0

### What engines are you seeing the problem on?

AMS

### How to reproduce

_No response_

### Relevant log output

```shell
2023-10-31 09:48:10,994 INFO [ASYNC-TableRuntimeRefreshExecutor-3] [org.apache.iceberg.SnapshotScan] [] - Scanning table com.netease.arctic.catalog.IcebergCatalogWrapper$BasicIcebergTable@4550db2b snapshot 1659655291600117103 created at 2023-10-31T01:47:36.277+00:00 with filter true
```

### Anything else

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by link3280**
## Why are the changes needed?
As titled.

Close #2255.

## Brief change log

-

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? no
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: core/src/main/java/com/netease/arctic/catalog/IcebergCatalogWrapper.java [hunk 1]
```diff
@@ -330,6 +332,11 @@ public BasicIcebergTable(
       super(tableIdentifier, icebergTable, arcticFileIO, catalogProperties);
     }
 
+    @Override
+    public String name() {
+      return icebergTable.name();
```
Comment (zhoujinsong) [inline core/src/main/java/com/netease/arctic/catalog/IcebergCatalogWrapper.java:337]: The superclass has implemented the `name` method, ref: https://github.com/NetEase/amoro/blob/master/core/src/main/java/com/netease/arctic/table/BasicUnkeyedTable.java#L119

The logging error is caused by some classes in Iceberg that directly use `Table#toString` to obtain the name of the table when logging, so it is necessary to implement the `toString` method in `BasicUnkeyedTable`, ref:https://github.com/apache/iceberg/blob/main/core/src/main/java/org/apache/iceberg/SnapshotScan.java#L126

Comment (link3280) [reply core/src/main/java/com/netease/arctic/catalog/IcebergCatalogWrapper.java:337]: OK. It seems that it's already fixed by #1914. But how about the absent serialization ID?

Comment (zhoujinsong) [reply core/src/main/java/com/netease/arctic/catalog/IcebergCatalogWrapper.java:337]: It seems that there won't be multiple versions of the class in the runtime environment, so it's acceptable to not set a serialization ID.

**Comment by link3280** [comment]
Fixed by https://github.com/NetEase/amoro/issues/1914 for future reference.

**Pull request closed without merging**